### PR TITLE
Issue3 display td tab

### PR DESCRIPTION
**Pull request by nipendar**
Should be good to merge...any issues identified could be fixed later as issue5 is a pre-requisite for Tab to fully work

### CODE REVIEW

File: app/controllers/concerns/foreman_teamdynamix/hosts_controller_extensions.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+module ForemanTeamdynamix
+  # Example: Plugin's HostsController inherits from Foreman's HostsController
+  module HostsControllerExtensions
+    extend ActiveSupport::Concern
+
+    def team_dynamix
+      find_resource
+      render partial: 'foreman_teamdynamix/hosts/team_dynamix', :locals => { :host => @host }
+    end
```
Comment (edestecd) [inline app/controllers/concerns/foreman_teamdynamix/hosts_controller_extensions.rb]: Why aren't we handling the ajax errors?
https://github.com/MiamiOH/foreman_custom_tab/blob/master/app/controllers/concerns/foreman_custom_tab/hosts_controller_extensions.rb#L12

File: app/helpers/concerns/foreman_teamdynamix/hosts_helper_extensions.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+module ForemanTeamdynamix
+  module HostsHelperExtensions
+    extend ActiveSupport::Concern
+    DEFAULT_TD_PANE_FIELDS = { 'Asset ID': 'ID', 'Owner': 'OwningCustomerName', 'Parent Asset': 'ParentID' }
+
+    def td_tab_title
```
Comment (edestecd) [inline app/helpers/concerns/foreman_teamdynamix/hosts_helper_extensions.rb]: def teamdynamix_title

File: app/controllers/concerns/foreman_teamdynamix/hosts_controller_extensions.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+module ForemanTeamdynamix
+  # Example: Plugin's HostsController inherits from Foreman's HostsController
+  module HostsControllerExtensions
+    extend ActiveSupport::Concern
+
+    def team_dynamix
```
Comment (edestecd) [inline app/controllers/concerns/foreman_teamdynamix/hosts_controller_extensions.rb]: teamdynamix (the plugin name does not have a _)

Comment (edestecd) [reply app/controllers/concerns/foreman_teamdynamix/hosts_controller_extensions.rb]: can you change everything to teamdynamix (the plugin name does not have a _)

There are several instances of team_dynamix

File: foreman_teamdynamix.gemspec [hunk 4]
```diff
@@ -16,4 +16,5 @@ Gem::Specification.new do |s|
   s.test_files = Dir['test/**/*']
   s.add_development_dependency 'rdoc'
   s.add_development_dependency 'rubocop'
+  s.add_dependency 'deface', '>=1.2.0', '< 2.0'
```
Comment (edestecd) [inline foreman_teamdynamix.gemspec]: This dependency version fails to install on our version of Foreman.
Please make your gemspec the same as this one: https://github.com/MiamiOH/foreman_custom_tab/blob/master/foreman_custom_tab.gemspec#L18

File: app/helpers/concerns/foreman_teamdynamix/hosts_helper_extensions.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+      SETTINGS[:team_dynamix][:title] || 'Team Dynamix'
+    end
+
+    def teamdynamix_fields(host)
+      td_pane_fields = SETTINGS[:team_dynamix][:fields] || DEFAULT_TD_PANE_FIELDS
+      @asset = TeamDynamixApi.get_asset(host.asset_id)
```
Comment (edestecd) [inline app/helpers/concerns/foreman_teamdynamix/hosts_helper_extensions.rb]: I think this field should be host.teamdynamix_asset_id so it is clear what plugin it comes from.
Also there will be many hosts without an associated assert at first so we should do something nicer if host.teamdynamix_asset_id is nil, like return this for fields or something [[_('Asset'), 'None Associated']]

**Comment by nipendar** [comment]
Done

**Pull request merged**